source/interface.cc

```diff
@@ -28,10 +29,55 @@ Interface<dim,spacedim,n_components,LAC>::
 set_time (const double &t) const
 {
   dirichlet_bcs.set_time(t);
+  // dirichlet_dot.set_time(t);
   forcing_terms.set_time(t);
   neumann_bcs.set_time(t);
+  // exact_solution.set_time(t);
 };
 
+template<int dim, int spacedim, int n_components, typename LAC>
+void
+Interface<dim,spacedim,n_components,LAC>::
+update_constraints (  const double &t,
+                      ConstraintMatrix &constraints,
+                      ConstraintMatrix &constraints_dot,
+                      const DoFHandler<dim,spacedim> &dof_handler) const
+{
+  set_time(t);
+
+  constraints.clear();
+  DoFTools::make_hanging_node_constraints (dof_handler, constraints);
+  dirichlet_bcs.interpolate_boundary_values(dof_handler, constraints);
+  constraints.close ();
+
+  // constraints_dot.clear();
+  // DoFTools::make_hanging_node_constraints (*dof_handler, constraints_dot);
+  // dirichlet_dot.interpolate_boundary_values(*dof_handler, constraints_dot);
+  // constraints_dot.close ();
+}
+
+template<int dim, int spacedim, int n_components, typename LAC>
+void
+Interface<dim,spacedim,n_components,LAC>::
+grid_estimate (  Triangulation<dim, spacedim> &tria,
+                 Vector<float> &estimated_error_per_cell,
+                 const DoFHandler<dim,spacedim> &dof_handler,
+                 unsigned int fe_degree,
+                 typename LAC::VectorType &distributed_solution) const
+{
+  std::vector<bool> mask(dim+1,true);
+  mask[dim] = false;
```

Comment:
Simply because I have no idea how to provide other error estimators by default, since deal.II has only one, and all other estimators must be written **manually**